node_modules/col-activities/lib/api.js

```diff
@@ -0,0 +1,275 @@
+ * @param  {User}             user                            The user earning activity points for the activity
+ * @param  {String}           type                            The type of the activity. One of the types in `col-activities/lib/constants.js`
+ * @param  {Number}           objectId                        The id of the object on which the activity is taking place (e.g., the asset id, the comment id, etc.)
+ * @param  {String}           objectType                      The type of the object on which the activity is taking place. One of `CollabosphereConstants.ACTIVITY.OBJECT_TYPES`
+ * @param  {User}             [actor]                         The user performing the activity when different than the user earning activity points
+ * @param  {Function}         callback                        Standard callback function
+ * @param  {Object}           callback.err                    An error that occurred, if any
+ * @param  {Activity}         callback.activity               The created activity
+ */
+var createActivity = module.exports.createActivity = function(course, user, type, objectId, objectType, actor, callback) {
+  // Parameter validation
+  var validationSchema = Joi.object().keys({
+    'type': Joi.any().valid(_.pluck(ActivitiesDefaults, 'type')).required()
+  });
+
+  var validationResult = Joi.validate({
+    'type': type
+  }, validationSchema);
+
+  if (validationResult.error) {
+    return callback({'code': 400, 'msg': validationResult.error.details[0].message});
+  }
+
+  // Create the activity in the DB
+  var activity = {
+    'course_id': course.id,
+    'user_id': user.id,
+    'type': type,
+    'object_type': objectType
+  };
+  if (objectType === 'asset') {
+    activity.asset_id = objectId;
+  } else {
+    activity.object_id = objectId;
+  }
+  if (actor) {
+    activity.actor = actor.id;
+  }
+
+  DB.Activity.create(activity).complete(function(err, activity) {
+    if (err) {
+      log.error({'err': err}, 'Failed to create a new activity');
+      return callback({'code': 500, 'msg': err.message});
+    }
+
+    // Retrieve the number of points that should be earned for
+    // the current activity in the current course
+    getActivityTypeConfiguration(course.id, function(err, configuration) {
+      if (err) {
+        return callback(err);
+      }
+
+      // Increase the points of the user
+      var points = _.findWhere(configuration, {'type': type}).points;
+      user.increment('points', {'by': points}).complete(function(err) {
+        if (err) {
+          log.error({'err': err}, 'Failed to increment the points for a user');
+          return callback({'code': 500, 'msg': err.message});
+        }
+
+        // If the current user is the user triggering the activity, update the timestamp
+        // at which the last activity took place for that user
+        if (!actor) {
+          user.update({'last_activity': new Date()}).complete(function(err) {
```

Comment:
No, `user.update` is correct as the timestamp at which the user performed the last activity is being updated